includes/admin/class-wc-admin-duplicate-product.php

```diff
@@ -215,6 +230,42 @@ private function get_product_to_duplicate( $id ) {
 
 		return $post;
 	}
+
+	/**
+	 * Generates a unique slug for a given product. We do this so that we can override the
+	 * behavior of wp_unique_post_slug(). The normal slug generation will run single
+	 * select queries on every non-unique slug, resulting in very bad performance.
+	 *
```

Comment:
Would you mind adding a since tag here so we know when the method was introduced?